pkg/onit/util.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2019-present Open Networking Foundation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package onit
+
+import (
+	"net"
+	"strconv"
+)
+
+// CheckFreePort checks if a port is available
+func CheckFreePort(port int, finished chan bool) {
```

Comment:
This can just return a `bool` unless we're planning to use it in another goroutine, which probably isn't necessary given we're using the result immediately to determine whether we can forward a port.